### PR TITLE
chore: Change testing to use cjs.

### PR DESCRIPTION
**Pull request by kinyoklion**
BEGIN_COMMIT_OVERRIDE
chore: Change browser packages testing to use CJS.
feat: Vendor escapeStringRegexp to simplify builds.
END_COMMIT_OVERRIDE

SDK-816

### CODE REVIEW

File: packages/sdk/browser/__tests__/BrowserDataManager.test.ts [hunk 1]
```diff
@@ -1,5 +1,4 @@
 import { jest } from '@jest/globals';
```
Comment (kinyoklion) [inline packages/sdk/browser/__tests__/BrowserDataManager.test.ts:1]: I put the TextEncoder and a crypto fix into a setup script.

File: packages/sdk/browser/jest.config.json
Comment (kinyoklion) [inline packages/sdk/browser/jest.config.json:1]: Converted to JSON to remove the decision of ESM/CJS.

File: packages/sdk/browser/package.json [hunk 2]
```diff
@@ -30,14 +30,12 @@
     "build": "tsc --noEmit && rollup -c rollup.config.js",
     "lint": "eslint . --ext .ts,.tsx",
     "prettier": "prettier --write '**/*.@(js|ts|tsx|json|css)' --ignore-path ../../../.prettierignore",
-    "test": "NODE_OPTIONS=--experimental-vm-modules npx jest --runInBand",
+    "test": "npx jest --runInBand",
```
Comment (kinyoklion) [inline packages/sdk/browser/package.json:33]: No longer require experimental options.

One of the many benefits is that I can get the debugger to work for this package now.

File: packages/sdk/browser/package.json [hunk 3]
```diff
@@ -62,6 +60,7 @@
     "prettier": "^3.0.0",
     "rimraf": "^5.0.5",
     "rollup": "^3.23.0",
+    "rollup-plugin-visualizer": "^5.12.0",
```
Comment (kinyoklion) [inline packages/sdk/browser/package.json:63]: This should have been a dev dependency to begin with.

File: packages/sdk/browser/package.json [hunk 4]
```diff
@@ -30,14 +30,12 @@
...
     "lint": "eslint . --ext .ts,.tsx",
     "prettier": "prettier --write '**/*.@(js|ts|tsx|json|css)' --ignore-path ../../../.prettierignore",
-    "test": "NODE_OPTIONS=--experimental-vm-modules npx jest --runInBand",
+    "test": "npx jest --runInBand",
     "coverage": "yarn test --coverage",
     "check": "yarn prettier && yarn lint && yarn build && yarn test"
   },
   "dependencies": {
-    "@launchdarkly/js-client-sdk-common": "1.10.0",
-    "escape-string-regexp": "^5.0.0",
```
Comment (kinyoklion) [inline packages/sdk/browser/package.json:39]: This package is ESM only, just vendoring it removes that as a factor in the builds.

File: packages/sdk/browser/setup-jest.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+const { TextEncoder, TextDecoder } = require('node:util');
+const crypto = require('node:crypto');
+
+global.TextEncoder = TextEncoder;
+
+Object.assign(window, { TextDecoder, TextEncoder });
+
+Object.defineProperty(global.self, "crypto", {
```
Comment (kinyoklion) [inline packages/sdk/browser/setup-jest.js:8]: This is to resolve the "digest of undefined" issue.

**Review by tanderson-ld — APPROVED** [review]

**Pull request merged**